### DISCUSSION TITLE
Defining a building.

### DISCUSSION THREAD

**Discussion by SerkanGuldal** [TUDelft-CNS-ATM/bluesky#285] [Q&A]
I am just learning BlueSky, and I need to simulate the flights close to ground level. Thus, I need to include buildings. Is it possible to define obstacles in BlueSky?
Thank you in advance.

**Comment by jooste** [comment]
Hi! This would currently be done with the areafilter functionality of BlueSky, where polygonal areas can be defined with the `POLY` stack command. Depending on what you want those buildings to mean to your traffic you have to implement additional functionality using these areas in a plugin yourself. If you have enough time you can also wait for the outputs of the Metropolis 2 project, which uses BlueSky to simulate an urban / high-density drone airspace setting.

**Reply by SerkanGuldal** [reply]
Hi,
I just need time to study what you suggested.
areafilter is imported by `from bluesky.tools import areafilter`. However, I could not find any explanation of how to do it or to get benefits. Is it possible to show a simple implementation?
Instead of POLY, POLYALT looks better since it is a 3D option. Of course, it will be a flight restricted zone with conflict resolution.

The other thing is that I tried to use `AREA`, but it deletes anything that goes outside or flies away. Some conflict resolution needs to be introduced to bounce back. If possible that would be useful to define "road". In general terms, is there such an option to define a road map?

**Reply by jooste** [reply]
Sorry for not replying to this anymore. I hope in the end you found out how to do this? There are several plugins in the master version that use areafilter polygons (AREA, SECTORCOUNT, METRICS, ..). If you are looking for an implementation of a street graph in BlueSky have a look at the Metropolis 2 fork.

**Reply by SerkanGuldal** [reply]
Thank you. I will check that out.

**Reply by SerkanGuldal** [reply]
Hello,
I had checked but no success. What I need is not a city graph too.
These define areas to be used for different purposes.
I might not explain well what I am trying to do. Let me put it as follows:
What I am trying is that areas be closed to flights (such as there is a building, or you can call a restricted zone). I don't want the flight to go through, even if it is on the way because of WPT definitions.
Maybe the resolution method activates and the area is not moveable, so the flight takes a turn to avoid entering the closed airspace.
Is this possible?

Here is an example view. The PLOYALT is high 1000. I need to fly around or over, not go through. 
![image](https://user-images.githubusercontent.com/9367866/163650773-91c5a759-c345-4e21-a382-568b0993d1e4.png)


Thank you for considering answering.

**Reply by jooste** [reply]
Hi @SerkanGuldal,

What you describe is not part of the core functions of bluesky. There are implementations of such rerouting algorithms, also implemented in bluesky in some of the forks, also in the metropolis 2 fork that I referred to previously. The recalculated route typically depends on some optimisation search, since there typically isn't a single unique solution to these rerouting problems. This also means that the implementations vary a lot.

**Reply by SerkanGuldal** [reply]
Thank you, for answering.
